ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -158,18 +160,31 @@ def getInputSpecification(cls):
                                                                         \item bitFlipMutator.
                                                                       \end{itemize}
                                                                     \end{itemize}
-                                                \item survivorSelectors:
+                                                  \item survivorSelectors:
                                                                       \begin{itemize}
                                                                         \item ageBased.
                                                                         \item fitnessBased.
                                                                       \end{itemize}
+                                                  \item constraintHandling:
+                                                                    \begin{itemize}
+                                                                      \item hard.
+                                                                      \item soft.
+                                                                    \end{itemize}          
                                                 \end{itemize}""")
     # Population Size
     populationSize = InputData.parameterInputFactory('populationSize', strictMode=True,
         contentType=InputTypes.IntegerType,
         printPriority=108,
         descr=r"""The number of chromosomes in each population.""")
     GAparams.addSub(populationSize)
+    
+    # Constraint Handling
+    constraintHandling = InputData.parameterInputFactory('constraintHandling', strictMode=True,
+        contentType=InputTypes.StringType,
+        printPriority=108,
+        descr=r"""a node indicating whether GA will handle constraints hardly or softly.""")
+    GAparams.addSub(constraintHandling)
+        
```

Comment:
is 'constraintHandling' used in handleInput?

Reply:
Constraint Handling is now commented and left for future improvement.